### PR TITLE
feat(financial-templates-lib): add price feed for yearn-style vaults

### PR DESCRIPTION
**Pull request by mrice32**
**Motivation**

To support badger Setts, a feed was needed to support yearn-style vaults with a `pricePerFullShare` method that determines how much a share of the vault is worth in the underlying token.

**Summary**

This price feed uses the getPricePerFull share method to determine how much a share is worth at any given timestamp. Note: this price feed only works for v1 yearn vaults and badger setts.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [x]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

N/A

### CODE REVIEW

File: packages/financial-templates-lib/test/truffle/BalancerPriceFeed.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ contract("BalancerPriceFeed.js", function(accounts) {
   beforeEach(async function() {
     dexMock = await BalancerMock.new({ from: owner });
 
-    // The UniswapPriceFeed does not emit any info `level` events.  Therefore no need to test Winston outputs.
+    // The BalancerPriceFeed does not emit any info `level` events.  Therefore no need to test Winston outputs.
```
Comment (chrismaree) [inline packages/financial-templates-lib/test/truffle/BalancerPriceFeed.js:25]: nice

File: yarn.lock [hunk 2]
```diff
@@ -3830,31 +3830,6 @@
   resolved "https://registry.yarnpkg.com/@protobufjs/utf8/-/utf8-1.1.0.tgz#a777360b5b39a1a2e5106f8e858f2fd2d060c570"
   integrity sha1-p3c2C1s5oaLlEG+OhY8v0tBgxXA=
 
-"@sendgrid/client@^6.5.5":
```
Comment (chrismaree) [inline yarn.lock:3833]: what's happening here? I dont see any modifications to the `package.json`

**Review by chrismaree — APPROVED** [review]
Looks good! now that this is done are you basically ready to pull all the pieces together and build the feed? what's still outstanding?

File: packages/financial-templates-lib/src/price-feed/VaultPriceFeed.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+const { BlockFinder } = require("./utils");
```
Comment (nicholaspai) [inline packages/financial-templates-lib/src/price-feed/VaultPriceFeed.js:2]: Awesome to see this re-use the `BlockFinder` created originally for the BalancerPriceFeed!

File: yarn.lock (hunk 2 above)
Comment (mrice32) [reply yarn.lock:3833]: I just ran yarn and got that. My guess is that a previous PR didn't update the lock file? I was confused too, so I discarded the changes and ran yarn again and it produced the same thing.

**Comment by mrice32** [comment]
> Looks good! now that this is done are you basically ready to pull all the pieces together and build the feed? what's still outstanding?

@chrismaree This should be enough for bBadger. I'm writing an LP feed for the Sushi LP identifier.

**Pull request merged**